sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala

```diff
@@ -47,75 +42,40 @@ case class BroadcastHashJoin(
     right: SparkPlan)
   extends BinaryNode with HashJoin with CodegenSupport {
```

Comment:
once we have this broadcast operator, is it possible to generalize this operator to just become HashJoin?

(don't need to have it in this pr)